src/gwt/src/org/rstudio/studio/client/workbench/views/terminal/events/CreateTerminalEvent.java

```diff
@@ -0,0 +1,50 @@
+
+/*
+ * CreateTerminalEvent.java
+ *
+ * Copyright (C) 2009-14 by RStudio, Inc.
+ *
+ * Unless you have received this program directly from RStudio pursuant
+ * to the terms of a commercial license agreement with RStudio, then
+ * this program is licensed to you under the terms of version 3 of the
+ * GNU Affero General Public License. This program is distributed WITHOUT
+ * ANY EXPRESS OR IMPLIED WARRANTY, INCLUDING THOSE OF NON-INFRINGEMENT,
+ * MERCHANTABILITY OR FITNESS FOR A PARTICULAR PURPOSE. Please refer to the
+ * AGPL (http://www.gnu.org/licenses/agpl-3.0.txt) for more details.
+ *
+ */
+
+package org.rstudio.studio.client.workbench.views.terminal.events;
+
+import com.google.gwt.event.shared.EventHandler;
+import com.google.gwt.event.shared.GwtEvent;
+
+/**
+ * Event to trigger creation of a terminal session.
+ */
+public class CreateTerminalEvent extends GwtEvent<CreateTerminalEvent.Handler>
```

Comment:
This should be a `CrossWindowEvent` since only the main window is going to handle it (but it could be invoked via menus or shortcuts in other windows). For now all that's needed here is to extend `CrossWindowEvent` rather than `GwtEvent`.